[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57888 [PyTorch Edge] bytecode version bump to v5 and enable share constant table
* #58182 [PyTorch Edge] share tensors in mobile with new api
* #57678 [torch.package/TorchScript] flag to gate allowance of TS serializaiton in torch.package
* #54894 [torch.Package/TorchScript] TS into torch.Package test cases
* #57573 [torch.package/TorchScript] logic to enable sharing of tensors on load
* #54893 [torch.Package/TorchScript] torch.Package python logic to save TorchScript
* #56299 [torch.Package/TorchScript] ScriptModuleSerializer add unified format
* **#55958 [torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter**
* #54891 [torch.Package/TorchScript] TS serialization importer to handle unified format

This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

Commit:
[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter

[ghstack-poisoned]

Lilyjjo commented on torch/csrc/jit/serialization/export.h:
```diff
@@ -53,6 +55,42 @@ TORCH_API std::string serialize_model_proto_to_string(
+  void writeByteCode(const Module& module, bool save_mobile_debug_info);
+  void writeArchive(const IValue& value, const std::string& archive_name);
+
+  caffe2::serialize::PyTorchStreamWriter& writer_;
```
Changed from `caffe2::serialize::PyTorchStreamWriter writer_` since the next PR in the stack can only pass in a reference

Lilyjjo commented on torch/csrc/jit/serialization/export_module.cpp:
```diff
@@ -372,232 +372,219 @@ void SetExportModuleMobileInfoConverter(
+  }
+}
 
-  void writeCode(const at::NamedTypePtr& root_type) {
```
writeCode was split into `convertTypes()` and `writeFiles()`

Lilyjjo commented on torch/csrc/jit/serialization/export.h:
```diff
@@ -53,6 +55,42 @@ TORCH_API std::string serialize_model_proto_to_string(
+      const Module& module,
+      const ExtraFilesMap& extra_files);
+  void writeByteCode(const Module& module, bool save_mobile_debug_info);
+  void writeArchive(const IValue& value, const std::string& archive_name);
```
Switched the ordering of args in `writeArchive()` since the next PR adds more string arguments

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

suo reviewed (approved):
stamping, assuming these changes are purely mechanical.

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27832546).

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27832546).

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27832546).

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27832546).

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27832546).

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27832546).

Commit:
Update on "[torch.Package/TorchScript] refactor ScriptModuleSerializer Exporter"


This PR refactors the existing ScriptModuleSerializer to be exposed to the public. Most of the code is the same, git just thinks it's different due to it being shifted over a white space. I commented on the actual changes that weren't due to the white space shifting

Differential Revision: [D27832546](https://our.internmc.facebook.com/intern/diff/D27832546)

[ghstack-poisoned]
